doCNA/Distribution.py

```diff
@@ -22,39 +22,64 @@ def __init__ (self, values, p_thr = 0.3, thr_z = 1.0):
 
         """
         assert len (values) > Consts.LENGTH_THRESHOLD, print ("Not enough data points to consider distribution.")
-        single_G_par = fit_single_G (np.sort(values), alpha = 0.01, r = 0.5)
+        
+        try: 
+            single_G_par = fit_single_G (np.sort(values), alpha = 0.01, r = 0.5)
+            values_used = 'all'
+        except:
+            single_G_par = fit_single_G (np.sort(np.unique(values)), alpha = 0.01, r = 0.5)
+            values_used = 'unique'
+        
+                   
         self.all_parameters = {}
         self.all_parameters['single'] = single_G_par
-                
+        self.all_parameters['single']['values_used'] = values_used
+        
         z = np.abs(values-single_G_par['m'])/single_G_par['s']
         string = list ('O' * len(values))
         for i in np.where ((z < thr_z))[0]:
             string[i] = 'B'
         
         self.all_parameters['single']['string'] = string
-                
+        
+        ##UGLY solution                
         if single_G_par['p'] < p_thr:
-            double_G_par = fit_double_G (np.sort(values), alpha = 0.01, r = 0.5)
-            self.key = 'double'
-            self.parameters = double_G_par
-            self.all_parameters ['double'] = double_G_par
-            z0 = np.abs(values-double_G_par['m'][0])/double_G_par['s'][0]
-            z1 = np.abs(values-double_G_par['m'][1])/double_G_par['s'][1]
-            #generate string
-            string = list ('O' * len(values))
-            #marking C population
-            for i in np.where ((z0 < z1)&(z0 < thr_z))[0]:
-                string[i] = 'C'
-            #marking D population
-            for i in np.where ((z1 < z0)&(z1 < thr_z))[0]:
-                string[i] = 'D'
-            self.parameters['string'] = ''.join(string)
-            self.all_parameters['double']['string'] = string
-            self.string = ''.join(string)
+            try:
```

Comment:
when does this happen? why would this raise an exception?

Reply:
Same reason as for single distribution.